GraphOperation.swift

```diff
@@ -0,0 +1,27 @@
+public protocol GraphOperation {
```

Comment:
Hmm, random thought: should we call this something like `TensorOperation` or `TFOpProtocol`? These operations (IIUC) are not necessarily graph operations (e.g. could be eager).

Reply:
I wanted this to be not specific for TF. I don't prefer `TFOpProtocol`, but would be OK with changing it to `TensorOperation`